c10/xpu/XPUFunctions.cpp

```diff
@@ -44,8 +44,17 @@ void enumDevices(std::vector<std::unique_ptr<sycl::device>>& devices) {
 }
 
 inline void initGlobalDevicePoolState() {
-  // Enumerate all GPU devices and record them.
-  enumDevices(gDevicePool.devices);
+  // Attempt to initialize XPU devices. If no device is found or the driver is
+  // not installed correctly, issue a warning message instead of raising an
+  // exception to avoid disrupting the user experience.
+  try {
+    // Enumerate all GPU devices and record them.
+    enumDevices(gDevicePool.devices);
+  } catch (const sycl::exception& e) {
+    TORCH_WARN(
+        "Failed to initialize XPU devices. Did you install the driver correctly?");
```

Comment:
The message is not informative. Could you help refine the message?

Reply:
Updated.